read_compat.ts

```diff
@@ -0,0 +1,115 @@
+import Plugin = require('./index');
+import fs = require('fs');
+import path = require('path');
+import quickTemp = require('quick-temp');
+import mapSeries = require('promise-map-series');
+import rimraf = require('rimraf');
+import symlinkOrCopy = require('symlink-or-copy');
+const symlinkOrCopySync = symlinkOrCopy.sync;
+
+import { TransformNodeInfo, Builder } from './interfaces';
+
+interface PluginWithDescription extends Plugin {
+  description?: string;
+}
+
+// Mimic how a Broccoli builder would call a plugin, using quickTemp to create
+// directories
+module.exports = class ReadCompat {
+  pluginInterface: TransformNodeInfo;
+  cachePath?: string;
+  inputPaths: string[];
+  inputBasePath!: string;
+  outputPath!: string;
+
+  private _priorBuildInputNodeOutputPaths: string[];
+  callbackObject: Builder;
+
+  constructor(plugin: PluginWithDescription) {
+    this.pluginInterface = plugin.__broccoliGetInfo__();
+
+    quickTemp.makeOrReuse(this, 'outputPath', this.pluginInterface.name);
+
+    if (this.pluginInterface.needsCache) {
+      quickTemp.makeOrReuse(this, 'cachePath', this.pluginInterface.name);
+    } else {
+      this.cachePath = undefined;
+    }
+
+    quickTemp.makeOrReuse(this, 'inputBasePath', this.pluginInterface.name);
+
+    this.inputPaths = [];
+    this._priorBuildInputNodeOutputPaths = [];
+
+    if (this.pluginInterface.inputNodes.length === 1) {
+      this.inputPaths.push(this.inputBasePath);
+      this._priorBuildInputNodeOutputPaths.push(this.inputBasePath);
+    } else {
+      for (let i = 0; i < this.pluginInterface.inputNodes.length; i++) {
+        this.inputPaths.push(path.join(this.inputBasePath, i + ''));
+      }
+    }
+
+    this.pluginInterface.setup(null, {
+      inputPaths: this.inputPaths,
+      outputPath: this.outputPath,
+      cachePath: this.cachePath,
+    });
+
+    this.callbackObject = this.pluginInterface.getCallbackObject();
+
+    if (plugin.description == null) {
+      plugin.description = this.pluginInterface.name;
+      if (this.pluginInterface.annotation != null) {
+        plugin.description += ': ' + this.pluginInterface.annotation;
+      }
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  read(readTree: any) {
```

Comment:
idea: 

I think `readTree` can be typed as a function that always returns a string. If we want to get fancy, `mapSeries` always returns a `Promise<T[]>` where `T` is the `T` of `readTree<T>` of its second argument. Which in this case, I think may always be a string